bundles/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/discovery/FreeboxDiscoveryService.java

```diff
@@ -149,15 +148,16 @@ public void onDataFetched(ThingUID bridge, List<FreeboxLanHost> lanHosts,
             // Network devices
             for (FreeboxLanHost host : lanHosts) {
                 String mac = host.getMAC();
-                if (StringUtils.isNotEmpty(mac)) {
+                if (mac != null && !mac.isEmpty()) {
                     if (discoverNetDevice) {
                         String uid = mac.replaceAll("[^A-Za-z0-9_]", "_");
                         thingUID = new ThingUID(FreeboxBindingConstants.FREEBOX_THING_TYPE_NET_DEVICE, bridge, uid);
-                        String name = StringUtils.isEmpty(host.getPrimaryName()) ? ("Freebox Network Device " + mac)
+                        String name = (host.getPrimaryName() == null || host.getPrimaryName().isEmpty())
+                                ? ("Freebox Network Device " + mac)
                                 : host.getPrimaryName();
                         logger.trace("Adding new Freebox Network Device {} to inbox", thingUID);
                         Map<String, Object> properties = new HashMap<>(1);
-                        if (StringUtils.isNotEmpty(host.getVendorName())) {
+                        if (host.getVendorName() != null && !host.getVendorName().isEmpty()) {
```

Comment:
I would prefer a local variable here instead of calling `getVendorName` three times. Also on similiar occurrences below

Reply:
You're absolutely right. Now fixed.